### PR TITLE
Add support for deploying with S3 sync

### PR DESCRIPTION
**Pull request by cnunciato**
This change represents the first of two steps in migrating to using`aws s3 sync` to manage deployments of the website. It is completely additive (new bucket, event handler and associated infrastructure) so it can be merged safely without disrupting the current deployment.

### CODE REVIEW

File: infrastructure/index.ts [hunk 1]
```diff
@@ -41,26 +47,77 @@ const contentBucket = new aws.s3.Bucket(
...
     {
         protect: false,
-    });
+    },
+);
 
-// contentBucket needs to have the "public-read" ACL so its contents can be ready by CloudFront and
-// served. But we deny the s3:ListBucket permission to prevent unintended disclosure of the bucket's
-// contents.
-const denyListPolicyState: aws.s3.BucketPolicyArgs = {
```
Comment (cnunciato) [inline infrastructure/index.ts]: Removing this to allow for `s3 ls`.

Comment (justinvp) [reply infrastructure/index.ts]: You wouldn't happen to know of a policy that would deny it from being listed by any rando, but would still allow us to list it internally in the AWS console?

Comment (cnunciato) [reply infrastructure/index.ts]: I do not, but I'm not yet an expert here, either. I can imagine there might be some way to deny all, but allow some assumed role, perhaps?

Comment (cnunciato) [reply infrastructure/index.ts]: We should be able to apply an exception to that `Deny` rule, so I've restored this block and will submit a separate PR with that alone. Thanks for the nudge on it. 😄

**Pull request merged**